go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -121,6 +133,26 @@ func main() {
 		select {
 		case s := <-exitCh:
 			klog.Infof("Received signal %s. Shutting down", s)
+			if s == syscall.SIGTERM {
```

Comment:
note: There are a lot of nested ifs that I am not a fan of. However since we shouldn't really care to error out if we fail in one of the steps, I did this. But if folks feel I need to split them out to make it eye-friendly I can do that as well.

Reply:
Yeah, I'd pull this out into a separate function, and perhaps collapse the ifs by doing stuff like:

```
if ovnkclient, err := util.NewOVNClientset(&config.Kubernetes); err != nil {
    return fmt.Errorf(xxx)
}

if nodeObj, err := ovnkclient.KubeClient.CoreV1().Nodes().Get(context.TODO(), node, metav1.GetOptions{}); err != nil {
    return fmt.Errorf(xxxx)
}

etc.
```

But this probably should also go into pkg/node/node.go as a goroutine that Start() calls. It can add 1 to the WaitGroup passed to Start() to make sure the node process doesn't exit until we've added the taint. Then it can "<-n.stopChan" and after that do all the taint addition. The benefit of that is you already have a KubeClient near the top of Start() that you cna use:

```
	if node, err = n.Kube.GetNode(n.name); err != nil {
		return fmt.Errorf("error retrieving node %s: %v", n.name, err)
	}
```

Reply:
makes sense. Let me do that.

Reply:
done.